pkg/azure/api/api.go

```diff
@@ -63,27 +63,19 @@ type MetricsAPI interface {
 	ObserveRateLimit(operation string, duration time.Duration)
 }
 
-func constructAuthorizer(cloudName, userAssignedIdentityID string) (autorest.Authorizer, error) {
+func constructAuthorizer(env azure.Environment, userAssignedIdentityID string) (autorest.Authorizer, error) {
 	if userAssignedIdentityID != "" {
-		env, err := azure.EnvironmentFromName(cloudName)
-		if err != nil {
-			return nil, err
-		}
-		msiEndpoint, err := adal.GetMSIVMEndpoint()
+		spToken, err := adal.NewServicePrincipalTokenFromManagedIdentity(env.ServiceManagementEndpoint, &adal.ManagedIdentityOptions{
+			IdentityResourceID: userAssignedIdentityID,
+		})
 		if err != nil {
 			return nil, err
 		}
 
-		spToken, err := adal.NewServicePrincipalTokenFromMSIWithUserAssignedID(msiEndpoint,
```

Comment:
Move to using `NewServicePrincipalTokenFromManagedIdentity` because the old method was deprecated.